src/H5.c

```diff
@@ -963,8 +963,12 @@ H5check_version(unsigned majnum, unsigned minnum, unsigned relnum)
         HDsnprintf(lib_str, sizeof(lib_str), "HDF5 library version: %d.%d.%d", H5_VERS_MAJOR, H5_VERS_MINOR,
                    H5_VERS_RELEASE);
         if (*substr) {
-            HDstrncat(lib_str, "-", (size_t)1);
```

Comment:
Don't use strncat with size equal to source.

Reply:
This might be shorter / simpler as:  
HDsnprintf(lib_str, sizeof(lib_str), "HDF5 library version: %d.%d.%d%s%s", H5_VERS_MAJOR, H5_VERS_MINOR, H5_VERS_RELEASE, (*substr ? "-" : ""), substr);

Instead of the previous or updated code.

Reply:
Sure, it probably makes more sense to switch to using snprintf where we're able to anyway, rather than having to manually manage strncats.

Reply:
Fixed